lib/GenomeFileUtil/core/FastaGFFUploaderScript.py

```diff
@@ -0,0 +1,678 @@
+                logger.info('Logging workspace error on save_objects: {}\n{}'.format(e.message, e.data))
+                raise
+
+    genome = dict()
+    genome["id"]=core_genome_name
+    genome["scientific_name"]=scientific_name
+    genome["assembly_ref"]=assembly_ref
+    genome["features"]=genome_features_list
+    genome["cdss"]=genome_cdss_list
+    genome["mrnas"]=genome_mrnas_list
+    genome["source"]=source
+    genome["domain"]="Eukaryota"
+    genome["genetic_code"]=1
+    genome["gc_content"]=assembly["gc_content"]
+    genome["dna_size"]=assembly["dna_size"]
+
+    if taxon_reference is not None:
+        genome["taxon_ref"] = taxon_reference
+        genome["taxonomy"]=taxonomy
+
+    UserMeta = dict()
+    UserMeta['Taxonomy']=taxonomy
+    UserMeta['Source']=source
+    UserMeta['Domain']="Eukaryota"
+    UserMeta['Source ID']=core_genome_name
+    UserMeta['Name']=scientific_name
+    UserMeta['Genetic code']=1;
+
+    UserMeta['GC content']=assembly["gc_content"]
+    UserMeta['Size']=assembly["dna_size"]
+    UserMeta['Number contigs']=assembly['num_contigs']
+
+    #id_source_version_array = core_genome_name.split("_")
+    #version = "_".join(id_source_version_array[2:])
+    #UserMeta['Version']=version
+
+    UserMeta['url']='https://transfer.api.globusonline.org/v0.10/endpoint/78312867-6d04-11e5-ba46-22000b92c6ec/ls?path=By_Organism_Name//P/PhytozomeV11/';
+
+    if(gff_handle_ref == None):
+        #Upload GFF to shock
+        token = os.environ.get('KB_AUTH_TOKEN') 
+
+        shock_info = script_utils.upload_file_to_shock(logger, shock_service_url, input_gff_file, token=token)
+        shock_id = shock_info["id"]
+        handles = script_utils.getHandles(logger, shock_service_url, handle_service_url, [shock_id], [gff_handle_ref], token)   
+        gff_handle_ref=handles[0]
+    genome['gff_handle_ref'] = gff_handle_ref
+
+#    genome_string = simplejson.dumps(genome, sort_keys=True, indent=4, ensure_ascii=False)
+#    genome_file = open("Bulk_Phytozome_Upload/"+core_genome_name+'.json', 'w+')
+#    genome_file.write(genome_string)
+#    genome_file.close()
+
+    #Provencance has a 1 MB limit.  We may want to add more like the accessions, but to be safe for now not doing that.
+    genome_provenance = [{"script": __file__, "script_ver": "0.1", "description": "Genome from GFF from %s" % (source)}]
+
+    pprint(UserMeta)
+
+    logger.info("Attempting Genome save for %s" % (core_genome_name))
+    genome_not_saved = True
+    while genome_not_saved:
+        try:
+            genome_info =  ws_client.save_objects({"workspace":workspace_name,
+                                                   "objects":[ { "type": "KBaseGenomes.Genome",
+                                                                 "data": genome,
+                                                                 "name": core_genome_name,
+#                                                                 "meta": UserMeta,
```

Comment:
well- I guess you don't pass along the user meta anyway, so there is a lot of code that you can remove.